src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -718,4 +718,24 @@ public boolean hasIndicatedSections(String courseId) throws EntityDoesNotExistEx
         }
         return archivedCourseIds;
     }
+
+    private CourseAttributes validateAndCreateCourseAttributes(String courseId,
+                                                               String courseName, String courseTimeZone)
+            throws InvalidParametersException {
+
+        FieldValidator validator = new FieldValidator();
+        String timeZoneErrorMessage = validator.getInvalidityInfoForCourseTimeZone(courseTimeZone);
+        if (!timeZoneErrorMessage.isEmpty()) {
+            CourseAttributes dummyCourse = CourseAttributes
+                    .builder(courseId, courseName, ZoneId.of("UTC"))
+                    .build();
+            List<String> errors = dummyCourse.getInvalidityInfo();
+            errors.add(timeZoneErrorMessage);
+            throw new InvalidParametersException(errors);
```

Comment:
I am not clear on what this if block is doing. This seems like a very roundabout way of handing an invalid timezone. Why is `getInvalidityInfo` only called when there is a `timeZoneErrorMessage`? What if there are other issues with the parameters, but the `courseTimeZone` field is correct?

Reply:
Yeah i should have explained this part more. If you follow the flow, parameters checking will be done in `CourseDb` by calling `CourseAtrributes.getInvalidityInfo`. However it can only check for id and name, because once a `CourseAtrributes` is created, its time zone is automatically valid. Hence I have to check for the timezone beforehand here. If it fails, i mimick the checking of the other fields to add to the list of errors. Otherwise I leave the checking of the other fields to `CourseDb`, as how it's originally done.

Reply:
Might want to add some brief explanation in code comments to avoid future confusion?

Reply:
Yep, will do.